File: app/src/main/java/com/example/beerhive/beerlist/BeerRepositoryImplementation.java
package com.example.beerhive.beerlist;

import android.util.Log;

import androidx.lifecycle.MutableLiveData;

import com.example.beerhive.network.BeerServiceApi;
import com.example.beerhive.network.BeerServiceApiClient;
import com.example.beerhive.network.model.BeerResponse;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class BeerRepositoryImplementation implements BeerRepository {

   private BeerServiceApi beerServiceApi;

    public BeerRepositoryImplementation() {
        this.beerServiceApi = BeerServiceApiClient.getInstance();
    }

    @Override
    public void loadBeerListFromNetwork(final BeerLoaderCallback beerLoaderCallback) {
        beerServiceApi.getListOfBeers().enqueue(new Callback<List<BeerResponse>>() {
            @Override
            public void onResponse(Call<List<BeerResponse>> call, Response<List<BeerResponse>> response) {
                if (response.isSuccessful() && response.body() != null){
                    beerLoaderCallback.onBeerListLoaded(response.body());
                }else {
                    beerLoaderCallback.onErrorOccurred();
                }
            }

            @Override
            public void onFailure(Call<List<BeerResponse>> call, Throwable t) {
                Log.d("Call Failure",t.getMessage());
                beerLoaderCallback.onErrorOccurred();
            }
        });
    }
}

File: app/src/main/java/com/example/beerhive/beerlist/BeerRepositoryImplementation.kt
package com.example.beerhive.beerlist

import com.example.beerhive.beerlist.BeerRepository.BeerLoaderCallback
import com.example.beerhive.network.BeerServiceApi
import com.example.beerhive.network.BeerServiceApiClient

class BeerRepositoryImplementation : BeerRepository {

    private val beerServiceApi: BeerServiceApi? = BeerServiceApiClient.beerServiceApi

    override fun loadBeerListFromNetwork(beerLoaderCallback: BeerLoaderCallback?) {
//        beerServiceApi?.listOfBeersAsync()?.enqueue(object : Callback<List<BeerResponse?>?> {
//            override fun onResponse(call: Call<List<BeerResponse?>?>, response: Response<List<BeerResponse?>?>) {
//                if (response.isSuccessful && response.body() != null) {
//                    beerLoaderCallback?.onBeerListLoaded(response.body())
//                } else {
//                    beerLoaderCallback!!.onErrorOccurred()
//                }
//            }
//
//            override fun onFailure(call: Call<List<BeerResponse?>?>, t: Throwable) {
//                Log.d("Call Failure", t.message)
//                beerLoaderCallback?.onErrorOccurred()
//            }
//        })
    }

}